packages/devextreme/js/__internal/grids/new/grid_core/columns_controller/columns_controller.ts

```diff
@@ -137,4 +138,80 @@ export class ColumnsController {
 
     return result;
   }
+
+  private readonly getActualColumnType = (value: unknown): 'number' | 'date' | 'datetime' | 'string' => {
+    if (typeof value === 'number') {
+      return 'number';
+    }
+
+    if (typeof value === 'string') {
+      if (!isNaN(Number(value))) {
+        return 'number';
+      }
+
+      const parsed = Date.parse(value);
+      if (!isNaN(parsed)) {
```

Comment:
Please, re-assure that this behavior is similar to datagrid's:
- does it really tries to parse value?
- does it really use the pattern that you wrote? Maybe it uses some option instead